test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -656,6 +658,168 @@ public async Task DataSource()
             Assert.That(conn.DataSource, Is.EqualTo($"tcp://{conn.Host}:{conn.Port}"));
         }
 
+        #region Server version
+
+        [Test]
+        public async Task PostgreSqlVersion_ServerVersion()
+        {
+            await using var c = new NpgsqlConnection(ConnectionString);
+
+            Assert.That(() => c.PostgreSqlVersion, Throws.InvalidOperationException
+                .With.Message.EqualTo("Connection is not open"));
+
+            Assert.That(() => c.ServerVersion, Throws.InvalidOperationException
+                .With.Message.EqualTo("Connection is not open"));
+
+            await c.OpenAsync();
+            var backendVersionString = (string)(await c.ExecuteScalarAsync("SHOW server_version"))!;
+
+            Assert.That(backendVersionString, Is.EqualTo(c.ServerVersion));
+            if (backendVersionString.Contains("beta") || backendVersionString.Contains("devel"))
+                Assert.That(backendVersionString, Does.Contain(c.PostgreSqlVersion.Major.ToString()));
+            else
+                Assert.That(backendVersionString, Does.Contain(c.PostgreSqlVersion.ToString()));
+        }
+
+        [TestCase("X13.0")]
+        [TestCase("13.")]
+        [TestCase("13.1.")]
+        [TestCase("13.1.1.")]
+        [TestCase("13.1.1.1.")]
+        [TestCase("13.1.1.1.1")]
+        public void ParseVersionFails(string versionString)
+            => Assert.That(() => TestDbInfo.ParseServerVersion(versionString), Throws.Exception);
+
+        [TestCaseSource(nameof(ParseVersionSucceedsData))]
+        public Version ParseVersionSucceeds(string versionString)
+            => TestDbInfo.ParseServerVersion(versionString);
+
+        static IEnumerable<TestCaseData> ParseVersionSucceedsData
+        {
+            get
+            {
+                yield return new TestCaseData("13.3")
```

Comment:
Total insignificant nit: this could be simplified to a bunch of [TestCase] attributes where the expected result is given as a string, e.g.:

```c#
[TestCase("13.3", ExpectedResult = "13.3")]
public string ParseVersionSucceeds(string versionString)
    => TestDbInfo.ParseServerVersion(versionString).ToString();
```